server/api/crud/artifacts.py

```diff
@@ -58,6 +58,12 @@ def store_artifact(
         # calculate the size of the artifact
         self._resolve_artifact_size(artifact, auth_info)
 
+        # TODO: remove this in 1.8.0
+        if mlrun.utils.helpers.is_legacy_artifact(artifact):
+            artifact = mlrun.artifacts.base.convert_legacy_artifact_to_new_format(
+                artifact
+            ).to_dict()
+
```

Comment:
I don't think this is needed. IIRC we only need `convert_legacy_artifact_to_new_format` for data migrations. no one should create a legacy artifact